native/packages/react-native-bpk-component-badge/src/BpkBadge.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React from 'react';
+import { View, ViewPropTypes } from 'react-native';
+import PropTypes from 'prop-types';
+import BpkText from 'react-native-bpk-component-text';
+import { getStyle } from './styles';
+
+export const BADGE_TYPES = {
+  success: 'success',
+  warning: 'warning',
+  destructive: 'destructive',
+  light: 'light',
+  inverse: 'inverse',
+  outline: 'outline',
+};
+
+export const BADGE_DOCKED_TYPES = {
+  left: 'left',
+  right: 'right',
+};
+
+export type Props = {
+  message: string,
+  docked: $Keys<typeof BADGE_DOCKED_TYPES>,
+  type: $Keys<typeof BADGE_TYPES>,
+  style: ?(Object | Array<Object>),
+};
+
+const BpkBadge = (props: Props) => {
+  const { message, docked, type, style: userStyle } = props;
+  const {
+    border: borderStyle,
+    view: innerStyle,
+    text: textStyle,
+    outline: outlineStyle,
+  } = getStyle(docked, type);
+
+  const outlineViewStyle =
+    type === BADGE_TYPES.outline ? [outlineStyle, borderStyle] : null;
+  const innerViewStyle =
+    type === BADGE_TYPES.outline ? innerStyle : [innerStyle, borderStyle];
+
+  return (
+    <View style={userStyle}>
```

Comment:
Coming in to this PR I expected to see a tree more resembling this:
```
<View>
  <BpkText style={textStyle} textStyle="xs">
    {message}
  </BpkText>
</View>
```

Home come the two extra wrapper views are required?

Reply:
that was the original implementation, but the outmost `View` is to allow external styling without messing with the component and the one immediately after is for dealing with the outline, and the few problems I had with it, I'll double check if the inner one is stll needed

Reply:
I checked and because we can use opacity the inner view is not needed anymore, thanks for spotting that, I'll update the PR